File: src/CCefView/CefQuery.cpp
#include "CefQuery.h"

class CCefQuery::Implementaion
{
public:
  int64_t id_ = -1;
  int error_ = 0;
  bool restult_ = false;

  std::string request_;
  std::string response_;
};

CCefQuery::CCefQuery()
  : pImpl_(std::make_unique<Implementaion>())
{}

CCefQuery::CCefQuery(const std::string& req, const int64_t query)
  : CCefQuery()
{
  pImpl_->id_ = query;
  pImpl_->request_ = req;
}

CCefQuery::~CCefQuery()
{
  pImpl_.reset();
}

const std::string&
CCefQuery::getRequest() const
{
  return pImpl_->request_;
}

const int64_t
CCefQuery::getId() const
{
  return pImpl_->id_;
}

const std::string&
CCefQuery::getResponse() const
{
  return pImpl_->response_;
}

const bool
CCefQuery::getResult() const
{
  return pImpl_->restult_;
}

const int
CCefQuery::getError() const
{
  return pImpl_->error_;
}

void
CCefQuery::setResponseResult(bool success, const std::string& response, int error /*= 0*/)
{
  pImpl_->restult_ = success;
  pImpl_->response_ = response;
  pImpl_->error_ = error;
}

File: src/DNCef/CefQuery.cs
using System;

namespace DNCef
{
    public partial class CefQuery
    {
        public long Id
        {
            get { return GetId(); }
        }

        public string Request
        {
            get { return GetRequest(); }
        }

        public string Response
        {
            get { return GetResponse(); }
        }

        public bool Result
        {
            get { return GetResult(); }
        }

        public int Error
        {
            get { return GetError(); }
        }

        internal CefQuery(IntPtr query)
        {
            _native = query;
        }

        ~CefQuery()
        {
            Dispose(false);
        }
    }
}
